shared/constants/methods-tags.ts

```diff
@@ -6,15 +6,15 @@
  * network for the dapp. It could also be that a method expects the
  * globally selected network to match some value in the request params itself.
  */
-export const methodsRequiringNetworkSwitch = [
-  'eth_sendTransaction',
-  'eth_sendRawTransaction',
-  'wallet_switchEthereumChain',
-  'wallet_addEthereumChain',
-  'wallet_watchAsset',
-  'eth_signTypedData_v4',
-  'personal_sign',
-];
+export const methodsRequiringNetworkSwitch = Object.freeze([
+  'eth_sendTransaction' as const,
+  'eth_sendRawTransaction' as const,
+  'wallet_switchEthereumChain' as const,
+  'wallet_addEthereumChain' as const,
+  'wallet_watchAsset' as const,
+  'eth_signTypedData_v4' as const,
+  'personal_sign' as const,
+]);
```

Comment:
I think this ought to do the trick instead of `as const` on each line:
```suggestion
] as const);
```

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/24415/commits/eefefded21d463e3f26f7729f3246db6c9f7de4e